### PR TITLE
fix: check for double definition

### LINKED ISSUE #15815: Regression on Managed Namespace Metadata Option
Checklist:

* [X] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [X] I've included steps to reproduce the bug.
* [X] I've pasted the output of `argocd version`.

**Describe the bug**

It looks like [this change](https://github.com/argoproj/argo-cd/pull/15488) may have caused a regression that causes a duplicate namespace issue on a single application if a namespace object is has been defined in a manifest along with the `managedNamespaceMetadata` option enabled of an application. This seems to be related to the `argocd.argoproj.io/sync-options": "ServerSideApply=true` annotation being applied to namespaces. This causes a resource conflict on the defined namespace object.

The docs [here](https://argo-cd.readthedocs.io/en/stable/user-guide/sync-options/#namespace-metadata) say this should be possible with a namespace object.

**To Reproduce**

1. Create a namespace kubernetes manifest without server side apply in an application.
2. Enable `managedNamespaceMetadata:` on an application.

**Expected behavior**

Merge metadata from `managedNamespaceMetadata` when a namespace object is defined in manifests. This used to work before version v2.8.4

**Workaround***

Remove `managedNamespaceMetadata` block from application and only use k8s namespace manifest.

**Version**

v2.8.4

### PR DESCRIPTION
**Pull request by blakepettersson**
As found in #13965 (and as a follow-up to #13999), we also need to define what happens if _both_ `managedNamespaceMetadata`  _and_ an Application manifest are defined for the same namespace.

The idea here is that if that happens, ~we emit an
`ApplicationConditionRepeatedResourceWarning`, and set the sync status to `Unknown`, since it's unclear what is supposed to happen.~ the `managedNs` doesn't get added to the list of `targetObjs` if there is already a source definition of the namespace.

Fixes https://github.com/argoproj/argo-cd/issues/15815

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.

### CODE REVIEW

File: controller/state.go [hunk 1]
```diff
@@ -453,6 +453,18 @@ func (m *appStateManager) CompareAppState(app *v1alpha1.Application, project *v1
 				LastTransitionTime: &now,
 			})
 		}
+
+		// If we reach this path, this means that a namespace has been both defined in Git, as well in the
+		// application's managedNamespaceMetadata. We want to emit a warning in such cases.
+		if isManagedNamespace(targetObj, app) {
+			conditions = append(conditions, v1alpha1.ApplicationCondition{
+				Type:               v1alpha1.ApplicationConditionRepeatedResourceWarning,
```
Comment (blakepettersson) [inline controller/state.go:461]: @crenshaw-dev I reused `ApplicationConditionRepeatedResourceWarning`, let me know if I should create a new status or if this is good enough.

**Comment by blakepettersson** [comment]
Looking at the test failure, and revisiting [the rationale](https://github.com/argoproj/argo-cd/pull/10672#issuecomment-1276237395) in #10672 for why the things are the way they are, it's probably better to ensure that the `managedNs` doesn't get added to the list of `targetObjs` if there is already a source definition of the namespace.

Will take a deeper look shortly.

**Comment by crenshaw-dev** [comment]
@blakepettersson should we add a test for this?

**Comment by evenh** [comment]
I would love a backport for this as we're currently on 2.8.4 🙏🏻

**Review by ishitasequeira — APPROVED** [review]
LGTM!!

**Comment by blakepettersson** [comment]
Thanks a lot @ishitasequeira! Seems like there is some issue when it got merged though, fixed by #16275

**Pull request merged**